crates/rbuilder-primitives/src/lib.rs

```diff
@@ -1226,6 +1226,8 @@ impl SimValue {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct SimulatedOrder {
     pub order: Order,
+    /// Flag indicating whether it's a system order.
+    pub is_system: bool,
```

Comment:
I think its better to add this to order metadata (to its property of the order), it feels a bit out of place in simulated order.  Ideally we remove system_recipient_allowlist from ctx and populate that field on the rpc level so order is already prepared.